### DISCUSSION TITLE
Set title using routeParams

### DISCUSSION THREAD

**Discussion by rtritto** [vikejs/vike#878] [Discussion] [answered]
### Use case
Create a title using @id in path (_routeParams.id_).

### Example
route: `/movie/@id`
title: `Title - ${@id}`

### Alternatives (?) to set title
- In `+route` file
- In `+onBeforeRender` file

**Comment by brillout** [comment]
You can already do that: https://vite-plugin-ssr.com/filesystem-routing.

**Reply by rtritto** [reply]
How can I pass _routeParams.id_ to _title_?

**Reply by brillout** [reply]
See `/examples/react-full-v1/`.

**Reply by rtritto** [reply, marked as answer]
Thanks!

In `onBeforeRender` function related to page, I fixed with:
```ts
  return {
    pageContext: {
      config: {
        title: `Title - ${pageContext.routeParams.id}`
      }
    }
  }
```

Then use `pageContext.config.title` in `/renderer/+onRenderHtml.tsx`.
